template/script.sh.erb

```diff
@@ -1,5 +1,10 @@
 #!/usr/bin/env bash
-
+<%
+r_versions = context.version.split(':')
+modules_to_load = r_versions.first
+has_alternate_r_version = r_versions.size == 2
+alternate_r_version = has_alternate_r_version ? r_versions.last : ''
```

Comment:
This could just be:

```ruby
modules_to_load, alternate_r_version = context.version.split(':')
```

and

```erb
module load <%= modules_to_load %>
<% if alternate_r_version %>
module switch <%= alternate_r_version %>
export RSTUDIO_WHICH_R=/usr/local/<%= alternate_r_version %>
<% end %>
```

The reason:

1. `nil` is a falsy value, which `alternate_r_version` will be if there is no colon
2. the assignment doesn't raise any exceptions if there is a mismatch and works as you would expect, so its safe. For example:

```
irb(main):024:0> mods ,ralt = "I/3 R/3:R/4".split(':')
=> ["I/3 R/3", "R/4"]
irb(main):025:0> mods
=> "I/3 R/3"
irb(main):026:0> ralt
=> "R/4"
irb(main):027:0> mods ,ralt = "I/3 R/3".split(':')
=> ["I/3 R/3"]
irb(main):028:0> mods
=> "I/3 R/3"
irb(main):029:0> ralt
=> nil
irb(main):030:0> mods ,ralt = "I/3 R/3:R/4:R/5".split(':')
=> ["I/3 R/3", "R/4", "R/5"]
irb(main):031:0> mods
=> "I/3 R/3"
irb(main):032:0> ralt
=> "R/4"
```


Disclaimer: in most code I normally am not a fan of writing methods that return nil and usually treat that as a "code smelll"; but it is okay here